src/settingsview.cpp

```diff
@@ -114,9 +114,14 @@ void SettingsView::browseMonitorDir()
         return;
     }
     if (confirmDialogMonitorDir(dir)) {
+        auto previousDir = KiwixApp::instance()->getSettingsManager()->getMonitorDir();
         KiwixApp::instance()->getSettingsManager()->setMonitorDir(dir);
         KiwixApp::instance()->getLibrary()->setMonitorDirZims(QStringList());
         ui->monitorClear->show();
+        KiwixApp::instance()->getWatcher()->addPath(dir);
+        if (previousDir != "") {
+            KiwixApp::instance()->getWatcher()->removePath(previousDir);
+        }
         KiwixApp::instance()->getLibrary()->asyncLoadMonitorDir(dir);
```

Comment:
Instead of exposing the directory watcher from `KiwixApp` (the `getWatcher()` method) you can encapsulate this entire operation in a new method `KiwixApp::setMonitorDir(QString path)`.